packages/x-data-grid/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -728,6 +728,7 @@ export const useGridColumnResize = (
           ...column,
           width: widthByField[column.field],
           computedWidth: widthByField[column.field],
+          flex: 0,
```

Comment:
> once the column's dimension is changed, the flex for this column is ignored.

This looks good to me.

(Nitpick) However, I'm wondering if this would be undesirable for some users.

I imagine a view where the fraction of sizes based on flex values is important (i.e. screen resize should keep the same width ratio as the original component, e.g. 1:2 in the added test component), but an accidental double-click on separator (auto size) removes that behavior.

Did you already consider making the new behavior opt-in/out?

P.S. It may not cover the majority of use cases anyway, so feel free to ignore this suggestion until we have a report on this in the community.

Reply:
Good point, I think in that case they don't want the column to be resizable at all, so `resizable: false` should work.

Reply:
Yeah, especially if we don't have a mixture of flex (fraction-sensitive) + regular columns.